src/Core/Managed/Shared/Channel/IDeepCloneable.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Microsoft.ApplicationInsights.Channel
+{
+    /// <summary>
+    /// Indicates that an object support deep cloning.
+    /// </summary>
+    /// <typeparam name="T">The type of the object that results from the cloning operation.</typeparam>
+    public interface IDeepCloneable<out T>
+    {
+        /// <summary>
+        /// Clones the object deeply, so that the original object and its clones share no state 
+        /// and can be modified independently.
+        /// </summary>
+        /// <returns>The cloned object.</returns>
+        T DeepClone();
```

Comment:
I think it makes more sense as a separate interface. There are internal data contract classes that benefit from having it. 

I am going to move it to DataContracts namespace. 

We could consider making the interface internal, let me know if you want me to pursue that.